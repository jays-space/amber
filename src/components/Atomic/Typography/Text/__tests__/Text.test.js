import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { Text } from "..";

describe("<Text />", () => {
  const string =
    "Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus exercitationem unde dicta expedita a facilis quo, ad inventore molestiae dolorum?";

  it("should render correctly", () => {
    render(<Text>{string}</Text>);

    const textElement = screen.getByTestId("paragraph");
    expect(textElement).toContainHTML("p");
    expect(textElement).toHaveTextContent(string);
    expect(textElement).toMatchSnapshot();
  });

  it("should render bolded string content", () => {
    render(<Text strong>{string}</Text>);

    const textElement = screen.getByTestId("paragraph");
    expect(textElement.classList).toContain("font-semibold");
    expect(textElement.classList).toContain("text-cyan-900");
  });
});
