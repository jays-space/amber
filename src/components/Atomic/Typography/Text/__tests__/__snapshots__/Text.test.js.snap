// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render correctly 1`] = `
<p
  class="tracking-wide  text-slate-700 undefined"
  data-testid="paragraph"
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus exercitationem unde dicta expedita a facilis quo, ad inventore molestiae dolorum?
</p>
`;
